feat: adds note-c platform mutexes (#4)
blues/note-espidf: README.md

Component config  --->  Notecard Configuration
```

## Thread Safety

The component automatically provides thread-safe access to the Notecard in multi-threaded FreeRTOS applications.
The underlying [note-c](https://github.com/blues/note-c) library protects the Notecard from concurrent access using internal mutexes, so no additional locking is required for normal use.

### I2C Bus Sharing

> If you have other I2C peripherals on the same bus as the Notecard, use the I2C mutex API to coordinate access:

Reviewer: ```suggestion
If you have other I2C peripherals on the same bus as the Notecard, register your I2C mutex with `note-c` using `NoteSetFnI2CMutex()` to minimize the time spent under lock.

 For your convenience, we have provided a default implementation of I2C mutex APIs to coordinate access (example shown below):
```

```c
#include "notecard.h"

// Access your I2C peripherals
notecard_lock_i2c();
i2c_master_transmit(my_peripheral_handle, data, len, timeout);
notecard_unlock_i2c();
```

This ensures the `note-c` won't attempt I2C communication while you're accessing your other peripherals.

> In order to enable/disable the provided I2C bus mutex, for example to use your own mutex, use `menuconfig`:

Reviewer: ```suggestion
In order to enable/disable the provided I2C bus mutex (e.g. when using your own mutex), use `menuconfig`:
```

Component config  --->  Notecard Configuration
```

## Thread Safety

The component automatically provides thread-safe access to the Notecard in multi-threaded FreeRTOS applications.
The underlying [note-c](https://github.com/blues/note-c) library protects the Notecard from concurrent access using internal mutexes, so no additional locking is required for normal use.

### I2C Bus Sharing

If you have other I2C peripherals on the same bus as the Notecard, register your I2C mutex with `note-c` using `NoteSetFnI2CMutex()` to minimize the time spent under lock.

>  For your convenience, we have provided a default implementation of I2C mutex APIs to coordinate access (example shown below):

Reviewer: ```suggestion
For your convenience, we have provided a default implementation of I2C mutex APIs to coordinate access (example shown below):
```
